### PR TITLE
Nix `rustc_hir` reexports in rustc::hir

### PR DESCRIPTION
**Pull request by Centril**
r? @Zoxc cc @Mark-Simulacrum

### CODE REVIEW

**Comment by Centril** [comment]
Seems to be passing now..., finally :)

File: src/librustc_mir/transform/check_unsafety.rs [hunk 1]
```diff
@@ -1,15 +1,14 @@
-use rustc_data_structures::fx::FxHashSet;
-
-use rustc::hir;
-use rustc::hir::def_id::DefId;
-use rustc::hir::Node;
+use rustc::hir::intravisit;
```
Comment (nagisa) [inline src/librustc_mir/transform/check_unsafety.rs]: Hmm, shouldn’t this be `rustc_hir::intravisit`?

Comment (nagisa) [reply src/librustc_mir/transform/check_unsafety.rs]: Double-checked, its in `rustc::hir` for now still. Weird that we still have `rustc_hir` and `rustc::hir`.

Comment (Centril) [reply src/librustc_mir/transform/check_unsafety.rs]: "Under construction" :P

`intravisit` depends on the HIR `Map`, which depends on query stuff, so we couldn't move it.

**Comment by Centril** [comment]
Raising the prio of this a bit because it's blocking other work and because it's bit-rotty.

@bors p=1

**Pull request merged**